### PR TITLE
[12.0][IMP] Fix unclickable overlapping features, importing geo data, opacity in map widget

### PR DESCRIPTION
**Pull request by tarteo**
* Order polygons based on their size so big ones are on the background making small ones selectable
* Add ir.fields.converter to make importing geo fields work
* Add opacity in the map field widget
* Adds alias in geo_search so searching on same table works

### CODE REVIEW

**Comment by tarteo** [comment]
I'm not sure why Runbot is saying it's failing. It seems to be working fine.

image

**Comment by yvaucher** [comment]
@tarteo for Runbot, it is a bit picky and warnings will cause a red status. Not blocking though.

File: base_geoengine/__manifest__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 {
     'name': 'Geospatial support for Odoo',
-    'version': '12.0.1.0.2',
+    'version': '12.0.1.1.0',
```
Comment (yvaucher) [inline base_geoengine/__manifest__.py]: You don't need to change the version. This is done when merged by @OCA-git-bot .

**Comment by yvaucher** [comment]
Runbot issue is fixed here: #240

File: base_geoengine/geo_ir/ir_fields_converter.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    @api.model
+    def _str_to_geo_polygon(self, model, field, value):
+        return value, []
+
+    _str_to_geo_point = _str_to_geo_polygon
+    _str_to_geo_line = _str_to_geo_polygon
+    _str_to_geo_multi_line = _str_to_geo_polygon
+    _str_to_geo_multi_point = _str_to_geo_polygon
+    _str_to_geo_multi_polygon = _str_to_geo_polygon
```
Comment (yvaucher) [inline base_geoengine/geo_ir/ir_fields_converter.py:16]: @tarteo this change is part of the commit "[IMP] Fix unclickable overlapping features, importing geo data, opacity in map widget"

Could you explain to me what it does? Should it go in a separate commit?
I guess it's for importing data. But, by which mean? csv?

Comment (tarteo) [reply base_geoengine/geo_ir/ir_fields_converter.py:16]: Yes, this tells Odoo how to convert the string value in the CSV to something the ORM field understands.

**Comment by yvaucher** [comment]
@tarteo something is failing on runbot when trying to open the GeoEngine Demo menu

```
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/api.py", line 1049, in get
    value = self._data[key][field][record._ids[0]]
KeyError: 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/fields.py", line 1004, in __get__
    value = record.env.cache.get(record, self)
  File "/.repo_requirements/odoo/odoo/api.py", line 1051, in get
    raise CacheMiss(record, field)
odoo.exceptions.CacheMiss: ('dummy.zip(1,).total_sales', None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 656, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 314, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 698, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 346, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 941, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 519, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/odoo-12.0/addons/web/controllers/main.py", line 904, in search_read
    return self.do_search_read(model, fields, offset, limit, domain, sort)
  File "/home/odoo/odoo-12.0/addons/web/controllers/main.py", line 926, in do_search_read
    offset=offset or 0, limit=limit or False, order=sort or False)
  File "/.repo_requirements/odoo/odoo/models.py", line 4595, in search_read
    result = records.read(fields)
  File "/.repo_requirements/odoo/odoo/models.py", line 2798, in read
    vals[name] = convert(record[name], record, use_name_get)
  File "/.repo_requirements/odoo/odoo/models.py", line 5123, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/.repo_requirements/odoo/odoo/fields.py", line 1008, in __get__
    self.determine_value(record)
  File "/.repo_requirements/odoo/odoo/fields.py", line 1121, in determine_value
    self.compute_value(recs)
  File "/.repo_requirements/odoo/odoo/fields.py", line 1075, in compute_value
    self._compute_value(records)
  File "/.repo_requirements/odoo/odoo/fields.py", line 1066, in _compute_value
    getattr(records, self.compute)()
  File "/home/odoo/build/OCA/geospatial/base_geoengine_demo/models/geo_npa.py", line 37, in _compute_ZIP_total_sales
    {'dummy.zip.the_geom': [('id', '=', rec.id)]})])
  File "/home/odoo/build/OCA/geospatial/base_geoengine/geo_model.py", line 159, in geo_search
    offset=offset, limit=limit, order=order)
  File "/home/odoo/build/OCA/geospatial/base_geoengine/geo_operators.py", line 121, in geo_search
    cr.execute(sql, where_clause_params)
  File "/.repo_requirements/odoo/odoo/sql_db.py", line 148, in wrapper
    return f(self, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/sql_db.py", line 225, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: invalid reference to FROM-clause entry for table "dummy_zip"
LINE 1: ...ailing_machine", dummy_zip AS _dummy_zip_ WHERE (("dummy_zip...
                                                             ^
HINT:  Perhaps you meant to reference the table alias "_dummy_zip_".
```

**Review by yvaucher — CHANGES_REQUESTED** [review]
Needs fixing as it is currently broken on runbot.

I checked with last runbot on branch 12.0, the view is opening correctly.

File: base_geoengine/geo_operators.py [hunk 3]
```diff
@@ -196,7 +196,7 @@ def get_geo_equal_sql(self, table, col, value, rel_col=None,
         else:
             base = self.geo_field.entry_to_shape(value, same_type=False)
             compare_to = "ST_GeomFromText('%s')" % (base.wkt,)
-        return " %s.%s = %s" % (table, col, compare_to)
+        return " _%s_.%s = %s" % (table, col, compare_to)
```
Comment (yvaucher) [inline base_geoengine/geo_operators.py:199]: No sure which one of those function is breaking the queries with demo module installed.

Comment (yvaucher) [reply base_geoengine/geo_operators.py:199]: ping @tarteo in case you missed this.

The current PR is breaking geoengine, I cannot merge it.
